src/ZODB/DB.py

```diff
@@ -645,6 +645,15 @@ def _(c):
         self._mvcc_storage.close()
         del self.storage
         del self._mvcc_storage
+        # clean up references to other DBs
+        self.databases.clear()
```

Comment:
I wonder if this should be `self.databases = {}` or even `self.databases = None`? That way if someone else has a reference to the databases dict (not hard if you're manually constructing a multi-db) it doesn't get cleared out from under them.